Admin/AclAdminExtension.php

```diff
@@ -44,16 +46,22 @@ class AclAdminExtension extends AdminExtension
     protected $roleHierarchy;
 
     /**
-     * @param SecurityContextInterface $securityContext
-     * @param Connection               $databaseConnection
-     * @param array                    $roleHierarchy
+     * @param SecurityContextInterface|TokenStorageInterface $tokenStorage
+     * @param Connection                                     $databaseConnection
+     * @param array                                          $roleHierarchy
      */
     public function __construct(
-        SecurityContextInterface $securityContext,
+        $tokenStorage,
         Connection $databaseConnection,
         array $roleHierarchy = array()
     ) {
-        $this->securityContext = $securityContext;
+        if (!$tokenStorage instanceof TokenStorageInterface && !$tokenStorage instanceof SecurityContextInterface) {
+            throw new \InvalidArgumentException('$securityContext must be an instanse of SecurityContextInterface or of TokenInterface '
+                    . 'but [' .  get_class($tokenStorage) . '] given.');
+            
+        }
+        
+        $this->securityContext = $tokenStorage;
```

Comment:
@meyerbaptiste ok) I replaced it: https://github.com/vedro-compota/CoopTilleulsAclSonataAdminExtensionBundle/blob/patch-2/Admin/AclAdminExtension.php